src/protojson/JobReadyMsg.cc

```diff
@@ -0,0 +1,116 @@
+#include "http/MetaModule.h"
+#include "http/RequestBodyJSON.h"
+#include "util/common.h"
+#include "util/TimeUtils.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+using namespace std;
+using namespace nlohmann;
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.protojson.JobReadyMsg");
+}  // namespace
+
+namespace lsst::qserv::protojson {
+
+JobReadyMsg::Ptr JobReadyMsg::create(std::string const& replicationInstanceId,
+                                     std::string const& replicationAuthKey, std::string const& workerIdStr,
+                                     std::string const& czarName, CzarIdType czarId, QueryId queryId,
+                                     UberJobId uberJobId, std::string const& fileUrl, uint64_t rowCount,
+                                     uint64_t fileSize) {
+    auto jrMsg = Ptr(new JobReadyMsg(replicationInstanceId, replicationAuthKey));
+    jrMsg->_workerId = workerIdStr;
+    jrMsg->_czarName = czarName;
+    jrMsg->_czarId = czarId;
+    jrMsg->_queryId = queryId;
+    jrMsg->_uberJobId = uberJobId;
+    jrMsg->_fileUrl = fileUrl;
+    jrMsg->_rowCount = rowCount;
+    jrMsg->_fileSize = fileSize;
+    return jrMsg;
+}
+
+JobReadyMsg::Ptr JobReadyMsg::createFromJson(nlohmann::json const& jsWReq,
+                                             std::string const& replicationInstanceId,
+                                             std::string const& replicationAuthKey) {
+    string const fName("JobReadyMsg::createFromJson");
+    LOGS(_log, LOG_LVL_DEBUG, fName);
+    try {
+        if (jsWReq["version"] != http::MetaModule::version) {
+            LOGS(_log, LOG_LVL_ERROR, fName << " bad version");
+            return nullptr;
+        }
+
+        // Presumably, if these were wrong, it wouldn't have gotten this far.
+        auto repliInstId = http::RequestBodyJSON::required<string>(jsWReq, "instance_id");
+        auto repliAuthKey = http::RequestBodyJSON::required<string>(jsWReq, "auth_key");
+
+        auto jrMsg = create(repliInstId, repliAuthKey);
+
+        jrMsg->_workerId = http::RequestBodyJSON::required<string>(jsWReq, "workerid");
+        jrMsg->_czarName = http::RequestBodyJSON::required<string>(jsWReq, "czar");
+        jrMsg->_czarId = http::RequestBodyJSON::required<CzarIdType>(jsWReq, "czarid");
+        jrMsg->_queryId = http::RequestBodyJSON::required<QueryId>(jsWReq, "queryid");
+        jrMsg->_uberJobId = http::RequestBodyJSON::required<UberJobId>(jsWReq, "uberjobid");
+        jrMsg->_fileUrl = http::RequestBodyJSON::required<string>(jsWReq, "fileUrl");
+        jrMsg->_rowCount = http::RequestBodyJSON::required<uint64_t>(jsWReq, "rowCount");
+        jrMsg->_fileSize = http::RequestBodyJSON::required<uint64_t>(jsWReq, "fileSize");
+        return jrMsg;
+    } catch (invalid_argument const& exc) {
+        LOGS(_log, LOG_LVL_ERROR, string("JobReadyMsg::createJson invalid ") << exc.what());
+    }
+    return nullptr;
+}
+
+json JobReadyMsg::serializeJson() {
```

Comment:
The name doesn't say what the method does - it converts the object into the `json` representation. Note that no serialization is happening here since the JSON object is the first-class citizen of the C++ type system. The serialization is happening when you are calling:
```
json::dump()
```
Hence a better name would be:
```
json JobReadyMsg::toJson()
```